### PR TITLE
refactor: [M3-8459] - Remove `patch-package` package

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

- Removes the `patch-package` package 📦 
- We haven't needed to patch in years, so I think it's worth cleaning this up 🧹 

## How to test 🧪

- Verify dependencies install and work as expected
- Verify all automated testing passes

## As an Author I have considered 🤔

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: docs/development-guide/01-repository-structure.md [hunk 1]
```diff
@@ -59,8 +59,6 @@ A few notable directories in the root level of the manager package:
   - end-to-end tests
 - **/e2e**
   - old end-to-end tests [deprecated]
-- **/patches**
-  - patches applied to dependencies via patch-package
```
Comment (coliu-akamai) [inline docs/development-guide/01-repository-structure.md:63]: Unrelated to this pr, but I noticed that this doc could use an update in general, maybe in a separate ticket:
- there are now four packages within the 'packages' directory
- inside the manager package, directories like /scripts don't exist anymore (seems like scripts was moved outside the packages directory, unless that's different?)
- events.ts and eventsPolling.ts are no longer files in manager/src's root

probably some others but those I just saw at a quick glance

**Review by coliu-akamai — APPROVED** [review]
✅ confirmed dependencies install didn't have errors/worked as expected
✅ confirmed automated tests (failing cypress test is already known issue)

🧹 🎉

File: docs/development-guide/01-repository-structure.md (hunk 1 above)
Comment (bnussman-akamai) [reply docs/development-guide/01-repository-structure.md:63]: Good call, I'll make a ticket for those doc updates

Comment (bnussman-akamai) [reply docs/development-guide/01-repository-structure.md:63]: Created M3-8464

**Review by carrillo-erik — APPROVED** [review]

**Pull request merged**